### ISSUE TITLE
deterministic std::random_device

### ISSUE THREAD

**Issue by RalfKubis** [mariusbancila/stduuid#6]
Took me a while to realize that I ran into:

https://sourceforge.net/p/mingw-w64/bugs/338/#ddd3
[#338 std::random_device not working properly]

which is explicitly mentioned in:
https://en.cppreference.com/w/cpp/numeric/random/random_device

"std::random_device may be implemented in terms of an implementation-defined pseudo-random number engine if a non-deterministic source (e.g. a hardware device) is not available to the implementation. In this case each std::random_device object may generate the same number sequence."

In other words: Each time I create a new instance of uuid_random_generator, it generates me the same UUIDs with MinGW - somewhat unexpected.

Probably one could change the implementation of basic_uuid_random_generator::basic_uuid_random_generator() to circumvent or escalate the issue or offer a hook.
Cppreference links to a "Community MinGW random_device support implementation" which could be an alternative.

**Comment by mariusbancila** [comment]
`basic_uuid_random_generator` must be supplied with a generator. It does create one itself. You need to do something like this:

```cpp
TEST_CASE("Test multiple default generators", "[gen][rand]")
{
   uuid id1;
   uuid id2;

   {
      std::random_device rd;
      auto seed_data = std::array<int, std::mt19937::state_size> {};
      std::generate(std::begin(seed_data), std::end(seed_data), std::ref(rd));
      std::seed_seq seq(std::begin(seed_data), std::end(seed_data));
      std::mt19937 generator(seq);

      id1 = uuids::uuid_random_generator{ generator }();
      REQUIRE(!id1.is_nil());
      REQUIRE(id1.version() == uuids::uuid_version::random_number_based);
      REQUIRE(id1.variant() == uuids::uuid_variant::rfc);
   }

   {
      std::random_device rd;
      auto seed_data = std::array<int, std::mt19937::state_size> {};
      std::generate(std::begin(seed_data), std::end(seed_data), std::ref(rd));
      std::seed_seq seq(std::begin(seed_data), std::end(seed_data));
      std::mt19937 generator(seq);

      id2 = uuids::uuid_random_generator{ generator }();
      REQUIRE(!id2.is_nil());
      REQUIRE(id2.version() == uuids::uuid_version::random_number_based);
      REQUIRE(id2.variant() == uuids::uuid_variant::rfc);
   }

   REQUIRE(id1 != id2);
}
```
